### PR TITLE
AB#1535 Small changes to enable integration with Kong

### PR DESCRIPTION
**Pull request by sharpedavid**

### CODE REVIEW

File: hnclient-v2/src/main/java/ca/bc/gov/hlth/hnclientv2/RetrieveAccessToken.java [hunk 1]
```diff
@@ -32,13 +32,13 @@ public RetrieveAccessToken(String tokenEndpoint, String scope, ClientAuthenticat
...
-    public synchronized AccessToken getToken() throws Exception {
+    public synchronized String getToken() throws Exception {
 
         // Reuse the token if the expiry time is more than a minute away
         if (Instant.now().toEpochMilli() + 60_000 < tokenExpiryTime) {
             logger.info(String.format("Using existing access token"));
-            logger.info(String.format("Access token: %s", accessToken.toJSONString()));
-            return accessToken;
+            logger.debug(String.format("Access token: %s", accessToken.toJSONString()));
+            return accessToken.toAuthorizationHeader();
```
Comment (sharpedavid) [inline hnclient-v2/src/main/java/ca/bc/gov/hlth/hnclientv2/RetrieveAccessToken.java:41]: We need this in order to add the "Bearer" keyword. 

In the route we use `.setHeader("Authorization").method(retrieveAccessToken)`, but this invocation doesn't allow us to modify the method result, so any modifications have to happen inside the method.

File: hnclient-v2/src/main/java/ca/bc/gov/hlth/hnclientv2/Route.java [hunk 2]
```diff
@@ -62,15 +62,15 @@ public void configure() throws Exception {
...
         renewKeys();
 
-
         from("netty:tcp://{{hostname}}:{{port}}")
                 .log("Retrieving access token")
                 .setHeader("Authorization").method(retrieveAccessToken)
+                .to("log:HttpLogger?level=DEBUG&showBody=true&showHeaders=true&multiline=true")
                 .log("Sending to HNSecure")
-                .to("http://{{hnsecure-hostname}}:{{hnsecure-port}}/{{hnsecure-endpoint}}")
+                .to("http://{{hnsecure-hostname}}:{{hnsecure-port}}/{{hnsecure-endpoint}}?throwExceptionOnFailure=false")
```
Comment (sharpedavid) [inline hnclient-v2/src/main/java/ca/bc/gov/hlth/hnclientv2/Route.java:70]: The HTTP component throws exceptions for non-200 results. We want to handle non-200 results (e.g. 401 missing access token).

File: hnclient-v2/src/main/java/ca/bc/gov/hlth/hnclientv2/Route.java [hunk 3]
```diff
@@ -62,15 +62,15 @@ public void configure() throws Exception {
...
                 .log("Received response from HNSecure")
                 .convertBodyTo(String.class)
-                .log("Response message: ${body}")
+                .to("log:HttpLogger?level=DEBUG&showBody=true&showHeaders=true&multiline=true")
```
Comment (sharpedavid) [inline hnclient-v2/src/main/java/ca/bc/gov/hlth/hnclientv2/Route.java:73]: Logging headers is an easy way to show that Kong is working.

File: mock-hnsecure/pom.xml [hunk 4]
```diff
@@ -51,10 +51,10 @@
       <artifactId>camel-bean</artifactId>
     </dependency>
 
-    <!-- validate OAuth access tokens -->
+    <!-- oAuth2 token -->
     <dependency>
       <groupId>com.nimbusds</groupId>
-      <artifactId>nimbus-jose-jwt</artifactId>
+      <artifactId>oauth2-oidc-sdk</artifactId>
```
Comment (sharpedavid) [inline mock-hnsecure/pom.xml:57]: oauth2-oidc-sdk depends on nimbus-jose-jwt. Using it lets us parse the token including the Bearer keywork. This is kind of interesting but it points out that the "Bearer" keyword is a part of the OAuth spec, not the JWT spec.

**Review by tschia — APPROVED** [review]
All good

**Pull request merged**